torchx/lib/torchx/backend.ex

```diff
@@ -1295,19 +1294,14 @@ defmodule Torchx.Backend do
 
   @impl true
   def inspect(%T{} = tensor, inspect_opts) do
-    result =
-      if device?(tensor, :cpu) do
-        binary = Torchx.to_blob(from_nx(tensor))
-        Nx.Backend.inspect(tensor, binary, inspect_opts)
-      else
-        "Tensors on the GPU cannot be inspected. Explicitly transfer the tensor by calling Nx.backend_transfer/1"
-      end
+    limit = if inspect_opts.limit == :infinity, do: :infinity, else: inspect_opts.limit + 1
 
-    maybe_add_signature(result, tensor)
+    tensor
+    |> to_binary(min(limit, Nx.size(tensor)))
```

Comment:
@josevalim I'm getting a segfault whenever the tensor is transferred to cuda:

```elixir
iex(1)> Torchx.device_available? :cuda
true
iex(2)> t = Nx.iota {3}
#Nx.Tensor<
  s64[3]
  Torchx.Backend(cpu)
  [0, 1, 2]
>
iex(3)> t |> Torchx.from_nx
from_nx/1    
iex(3)> t |> Torchx.from_nx()
{:cpu, #Reference<0.793727343.3884580880.186395>}
iex(4)> t |> Torchx.from_nx() |> Torchx.to_device(:cuda)
{:cuda, #Reference<0.793727343.3884580886.186862>}
iex(5)> t |> Torchx.from_nx() |> Torchx.to_device(:cuda) |> Torchx.to_nx
[1]    22011 segmentation fault (core dumped)  iex -S mix
```

and

```elixir
iex(1)> t = Nx.iota({3}, backend: {Torchx.Backend, device: :cuda}); nil
nil
iex(2)> t = Nx.iota({3}, backend: {Torchx.Backend, device: :cuda})     
[1]    23015 segmentation fault (core dumped)  iex -S mix
```

Reply:
This also happens on main, though, on `to_binary`:

```elixir
iex(1)> t = Nx.iota({3}, backend: {Torchx.Backend, device: :cuda}); nil
nil
iex(2)> t
#Nx.Tensor<
  s64[3]
  Torchx.Backend(cuda)
  Tensors on the GPU cannot be inspected. Explicitly transfer the tensor by calling Nx.backend_transfer/1
>       
iex(3)> t |> Nx.to_binary
[1]    24818 segmentation fault (core dumped)  iex -S mix
```

Reply:
Hrm... is to_blob invalid for GPU devices then? Maybe we need to transfer them to the CPU first? Although I thought the C implementation would take care of that.